pyogrio/arrow.py

```diff
@@ -0,0 +1,13 @@
+import pyarrow as pa
+
+
+class OGRRecordBatchStreamReader(pa.RecordBatchStreamReader):
+    _geo_metadata: dict
+
+    @property
+    def geo_metadata(self):
+        return self._geo_metadata
+
+    @geo_metadata.setter
+    def geo_metadata(self, value):
+        self._geo_metadata = value
```

Comment:
Any comments on this general structure? Should this work?

Reply:
@jorisvandenbossche do you know if there's an issue with subclassing `pa.RecordBatchStreamReader` like this?

Reply:
Sorry for the slow response here! Will take a closer look tomorrow. Is there a specific issue you are seeing with subclassing, or just wondering in general if this is OK to do?

Reply:
I reverted the wrapper for subclassing